modules/mobile-agents/android/client/plugins/IDPProxy/src/org/wso2/mdm/agent/proxy/utils/ServerUtilities.java

```diff
@@ -0,0 +1,375 @@
+				message = "Invalid client protocol.";
+				throw new IDPTokenManagerException(message, e);
+			} catch (IOException e) {
+				responseParams.put(Constants.SERVER_RESPONSE_BODY, "Internal Server Error");
+				responseParams.put(Constants.SERVER_RESPONSE_STATUS,
+				                   Constants.INTERNAL_SERVER_ERROR);
+				message = "Server connectivity failure.";
+				throw new IDPTokenManagerException(message, e);
+			}
+		}
+
+		return responseParams;
+	}
+
+	public static HttpClient getCertifiedHttpClient() throws IDPTokenManagerException {
+		HttpClient client = null;
+		InputStream inStream = null;
+		String message;
+		try {
+			if (Constants.SERVER_PROTOCOL.equalsIgnoreCase("https://")) {
+				KeyStore localTrustStore = KeyStore.getInstance("BKS");
+				inStream =
+						IdentityProxy.getInstance().getContext().getResources()
+						             .openRawResource(R.raw.emm_truststore);
+				localTrustStore.load(inStream, Constants.TRUSTSTORE_PASSWORD.toCharArray());
+
+				SchemeRegistry schemeRegistry = new SchemeRegistry();
+				schemeRegistry.register(new Scheme("http", PlainSocketFactory.getSocketFactory(),
+				                                   80));
+				SSLSocketFactory sslSocketFactory = new SSLSocketFactory(localTrustStore);
+				sslSocketFactory.setHostnameVerifier(SSLSocketFactory.ALLOW_ALL_HOSTNAME_VERIFIER);
+				schemeRegistry.register(new Scheme("https", sslSocketFactory, 443));
+				HttpParams params = new BasicHttpParams();
+				ClientConnectionManager connectionManager =
+						new ThreadSafeClientConnManager(params,
+						                                schemeRegistry);
+
+				client = new DefaultHttpClient(connectionManager, params);
+
+			} else {
+				client = new DefaultHttpClient();
+			}
+
+		} catch (KeyStoreException e) {
+			message = "Invalid keystore.";
+			throw new IDPTokenManagerException(message, e);
+		} catch (CertificateException e) {
+			message = "Invalid certificate.";
+			throw new IDPTokenManagerException(message, e);
+		} catch (NoSuchAlgorithmException e) {
+			message = "Keystore algorithm does not match.";
+			throw new IDPTokenManagerException(message, e);
+		} catch (UnrecoverableKeyException e) {
+			message = "Invalid keystore.";
+			throw new IDPTokenManagerException(message, e);
+		} catch (KeyManagementException e) {
+			message = "Invalid keystore.";
+			throw new IDPTokenManagerException(message, e);
+		} catch (IOException e) {
+			message = "Trust store failed to load.";
+			throw new IDPTokenManagerException(message, e);
+		} finally {
+			StreamHandler.closeInputStream(inStream, TAG);
+		}
+
+		return client;
+	}
+
+	public static String getResponseBody(HttpResponse response) throws IDPTokenManagerException {
+
+		String responseBody = null;
+		HttpEntity entity = null;
+		String message;
+
+		try {
+			entity = response.getEntity();
+			responseBody = getResponseBodyContent(entity);
+		} catch (ParseException e) {
+			message = "Invalid response.";
```

Comment:
Here, the assigning the log message to a variable called "message" is redundant, I believe. Instead, you can directly pass the message into the Exception instance.